### ISSUE TITLE
ag just printing "truncated file: Success"  when searching ~120M gzip files

### ISSUE THREAD

**Issue by Avinm** [ggreer/the_silver_searcher#1243]
Running `ag -z "searchStr" file.gz` results in just the output `ag: truncated file: Success` on some Unix systems (3.2.45-0.6 x86_64). Running the same search on the same file on OSX yields matches and line numbers as expected. What could be the issue?

**Comment by Gautham** [comment]
I've noticed this happens with ag-2.1 but not with ag-2.0.

**Comment by bjeanes** [comment]
I'm also experiencing this. I will attempt to downgrade to 2.0 to see if it resolves the issue for me.

**Comment by bjeanes** [comment]
Yep, 2.0 works. FWIW I was working off a build from master (as of today) building as per https://gist.github.com/steamedbuninvasion/68734672222ea181ca27e589a3ea6587 for AWS Linux. Doing a `git checkout 2.0.0` and re-building resulted in a binary that searched compressed files correctly.

**Comment by halloleo** [comment]
I still experience this as well: I use version 2.1 on CentOS 6.

**Comment by Avinm** [comment]
@halloleo : I switched to using [ripgrep](https://github.com/BurntSushi/ripgrep) to work around this. Seems to be more actively maintained.

**Comment by GarrettAlbright** [comment]
Still happening with version 2.2.0 on Ubuntu bullseye/sid.

**Comment by Mazorius** [comment]
> Still happening with version 2.2.0 on Ubuntu bullseye/sid.

Also running in that issue on Debian 10
